File: src/listas/java/List.java
package listas.java;

import java.util.Scanner;

public class List {
    private Node list;

    public List() {
        list = null;
    }

    public void insert(String auxId) {
        if (search(auxId) == null) {
            Node newNode = new Node(auxId);

            if (list == null) {
                list = newNode;
            } else {
                Node aux = list;
                list = newNode;
                list.setNext(aux);
            }
        } else {
            System.out.println("This ID already exists.");
        }
    }

    public void show() {
        if (list != null) {
            Node current = list;

            while (current != null) {
                System.out.println("the id: " + current.getId());

                current = current.getNext();
            }
        } else {
            System.out.println("The list is empty.");
        }
    }

    public void update(String auxId) {
        if (list != null) {
            Node searchId = search(auxId);

            if (searchId != null) {
                Scanner sc = new Scanner(System.in);

                System.out.println("Insert the new Id:");
                String auxNew = sc.nextLine();

                if (search(auxNew) != null) {
                    searchId.setId(auxNew);
                } else {
                    System.out.println("Already exist a person with this ID: $auxNew.");
                }
            } else {
                System.out.println("Exist no one person with the ID: $auxId.");
            }
        } else {
            System.out.println("The list is empty.");
        }
    }

    public void remove(String auxId) {
        if (list != null) {
            Node searchId = search(auxId);

            if (searchId != null) {
                if (searchId == list) {
                    list = list.getNext();
                } else {
                    Node current = list;
                    Node before = null;

                    while (current != null) {
                        if (auxId.equals(current.getId())) {
                            assert before != null;
                            before.setNext(current.getNext());
                            break;
                        }

                        before = current;
                        current = current.getNext();
                    }
                }
            } else {
                System.out.println("Exist no one person with de ID: $auxId");
            }
        } else {
            System.out.println("The list is empty.");
        }
    }

    private Node search(String auxId) {
        Node current = list;

        while (current != null) {
            if (auxId.equals(current.getId())) {
                return current;
            }

            current = current.getNext();
        }

        return null;
    }

}

File: src/listas/kotlin/List.kt
package listas.kotlin

class Node(var id: String) {
    var next: Node? = null
}

class List {
    private var list: Node? = null

    fun insert(auxId: String) {
        if (search(auxId) == null) {
            val newNode = Node(auxId)

            if (list == null) {
                list = newNode
                list!!.next = null
            } else {
                val aux = list
                list = newNode
                list!!.next = aux
            }
        } else {
            println("This ID already exists.")
        }
    }

    fun show() {
        if (list != null) {
            var current = list

            while (current != null) {
                println("The id: ${current.id}")

                current = current.next
            }
        } else {
            println("The list is empty.")
        }
    }

    fun update(auxId: String) {
        if (list != null) {
            val searchId = search(auxId)

            if (searchId != null) {
                println("Insert the new Id:")
                val auxNew = readLine()!!

                if (search(auxNew) != null) {
                    searchId.id = auxNew
                } else {
                    println("Already exist a person with this ID: $auxNew.")
                }
            } else {
                println("Exist no one person with the ID: $auxId.")
            }
        } else {
            println("The list is empty.")
        }
    }

    fun remove(auxId: String) {
        if (list != null) {
            val searchId = search(auxId)

            if (searchId != null) {
                if (searchId == list) {
                    val aux: Node? = list!!.next
                    list = aux
                } else {
                    var current = list
                    var before: Node? = null

                    while (current != null) {
                        if (auxId == current.id) {
                            before!!.next = current.next
                            break
                        }

                        before = current
                        current = current.next
                    }
                }
            } else {
                println("Exist no one person with de ID: $auxId")
            }
        } else {
            println("The list is empty.")
        }
    }

    private fun search(auxId: String): Node? {
        var current = list

        while (current != null) {
            if (auxId == current.id) {
                return current
            }

            current = current.next
        }

        return null
    }
}